docs/spec/spec.md

```diff
@@ -0,0 +1,202 @@
+# Object Model
+
+Knative ships with several objects which implement useful sets of
+[interfaces](interfaces.md). It is expected that additional objects which
+implement other sets of interfaces will be added over time. For context, see
+the [overview](overview.md) and [motivations](motivation.md) sections.
+
+These are Kubernetes resources that been introduced using Custom Resource
+Definitions. They will have the expected _ObjectMeta_, _Spec_, _Status_ fields.
+This document details our _Spec_ and _Status_ customizations.
+
+- [Channel](#kind-channel)
+- [Subscription](#kind-subscription)
+- [Provider](#kind-provisioner)
+
+## kind: Channel
+
+### group: eventing.knative.dev/v1alpha1
+
+_A Channel logically receives events on its input domain and forwards them to
+its subscribers. Additional behavior may be introduced by using the
+Subscription's call parameter._
+
+### Object Schema
+
+#### Spec
+
+| Field         | Type                               | Description                                                                | Limitations |
+| ------------- | ---------------------------------- | -------------------------------------------------------------------------- | ----------- |
+| provisioner\* | ProvisionerReference               | The name of the provisioner to create the resources that back the Channel. | Immutable.  |
+| arguments     | runtime.RawExtension (JSON object) | Arguments to be passed to the provisioner.                                 |             |
+
+\*: Required
```

Comment:
you have said this before and I do not agree it is readable because we have one-ofs that are required and some that are optional. It is way more clear to use a lookup table.